src/service/xhr-impl.js

```diff
@@ -94,6 +96,15 @@ export class Xhr {
   constructor(win) {
     /** @const {!Window} */
     this.win = win;
+
+    // TODO(@jridgewell, #9048): Once we figure out if ampdoc can always be
+    // found, remove the try catch.
+    try {
+      this.ampdoc_ = ampdocServiceFor(win).getAmpDoc();
+    } catch (e) {
+      this.ampdoc_ = null;
+      dev().error('XHR', e, 'XHR service failed to find ampdoc');
```

Comment:
The only way to pull the parent ampdoc is by passing around the context node (element). The reason why I think PWA mode is less relevant b/c this is a privacy feature, since iframes can pull documents from different origins. PWA on the other hand is built to serve "own" same-origin documents. If PWA decides to request cross-origin docs, it still needs to support CORS, etc - which essentially nullify our protection here.